// Copyright 2018 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package minigo;

// TODO(tommadams): Investigate enabling arenas with:
//   option cc_enable_arenas = true;

service InferenceService {
  rpc GetConfig(GetConfigRequest) returns (GetConfigResponse) {
  }
  rpc GetFeatures(GetFeaturesRequest) returns (GetFeaturesResponse) {
  }
  rpc PutOutputs(PutOutputsRequest) returns (PutOutputsResponse) {
  }
}

message GetConfigRequest {
}

message GetConfigResponse {
  int32 board_size = 1;
  int32 batch_size = 2;
}

message GetFeaturesRequest {
}

message GetFeaturesResponse {
  int32 batch_id = 1;
  repeated float features = 2;
}

message PutOutputsRequest {
  int32 batch_id = 1;
  repeated float policy = 2;
  repeated float value = 3;
}

message PutOutputsResponse {
}
